backend/recipes/base.py

```diff
@@ -293,15 +288,41 @@ def areTasksComplete(self):
         False
             Some tasks are still running.
         """
-        return all(task.ready() for task in self.currentTasks)
+        return all(task["done"] for task in self.currentTasks)
 
     def stopTasks(self):
         """
         Stop the currently running tasks.
         :return:
             None
         """
-        for task in self.currentTasks:
-            task.revoke(terminate=True)
+        # for task in self.currentTasks:
+        #     task.cancel()
         self.currentTasks = []
 
+    def tickTasks(self):
+        """
+        Executes one iteration of the current tasks that are scheduled to run.
+        :return:
+            None
+        """
+        for task in self.currentTasks:
+            if not task["done"] and datetime.now() > task["nextTime"]:
+                print("task is ready for next iteration")
```

Comment:
Can we remove this (and other fluffy print statements that would appear a lot in the logs)? Errors and whatnot are great but I've filled up a lot of disk space accidentally when I left my testing unit on for an extended period with some extraneous debugging print.